### PR TITLE
Sorry for late. Forgot click "create pull request" in request preview. Please look at commit time - it was before 23.59

### PR DESCRIPTION
**Pull request by skipor**

### CODE REVIEW

**Comment by thevery** [comment]
photos are lost on screen rotate

**Comment by thevery** [comment]
refresh in action bar is nice

**Comment by thevery** [comment]
second screen photos are not loaded at all because of OOM:
E/AndroidRuntime(25724): Process: ru.skipor.pohularPhotos, PID: 25724
E/AndroidRuntime(25724): java.lang.RuntimeException: An error occured while executing doInBackground()
E/AndroidRuntime(25724):    at android.os.AsyncTask$3.done(AsyncTask.java:300)
E/AndroidRuntime(25724):    at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:355)
E/AndroidRuntime(25724):    at java.util.concurrent.FutureTask.setException(FutureTask.java:222)
E/AndroidRuntime(25724):    at java.util.concurrent.FutureTask.run(FutureTask.java:242)
E/AndroidRuntime(25724):    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
E/AndroidRuntime(25724):    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
E/AndroidRuntime(25724):    at java.lang.Thread.run(Thread.java:841)
E/AndroidRuntime(25724): Caused by: java.lang.OutOfMemoryError
E/AndroidRuntime(25724):    at android.graphics.BitmapFactory.nativeDecodeStream(Native Method)
E/AndroidRuntime(25724):    at android.graphics.BitmapFactory.decodeStreamInternal(BitmapFactory.java:613)
E/AndroidRuntime(25724):    at android.graphics.BitmapFactory.decodeStream(BitmapFactory.java:589)
E/AndroidRuntime(25724):    at android.graphics.BitmapFactory.decodeStream(BitmapFactory.java:627)
E/AndroidRuntime(25724):    at ru.skipor.Utils.InternalStorageUtils.loadBitmap(InternalStorageUtils.java:142)
E/AndroidRuntime(25724):    at ru.skipor.Utils.InternalStorageUtils.loadOrDownloadAndSaveBitmap(InternalStorageUtils.java:55)
E/AndroidRuntime(25724):    at ru.skipor.popularPhotos.FullImageActivity$LoadImageTask.doInBackground(FullImageActivity.java:64)
E/AndroidRuntime(25724):    at ru.skipor.popularPhotos.FullImageActivity$LoadImageTask.doInBackground(FullImageActivity.java:60)
E/AndroidRuntime(25724):    at android.os.AsyncTask$2.call(AsyncTask.java:288)
E/AndroidRuntime(25724):    at java.util.concurrent.FutureTask.run(FutureTask.java:237)
E/AndroidRuntime(25724):    ... 3 more
W/ActivityManager(  765):   Force finishing activity ru.skipor.pohularPhotos/ru.skipor.popularPhotos.FullImageActivity

**Comment by thevery** [comment]
action bar on the second screen is useless

**Comment by thevery** [comment]
second screen images seems to be lazily loaded

**Comment by thevery** [comment]
app is ok (20-8=12), code review pending

File: pohularPhotos/src/main/java/ru/skipor/Utils/HTTPUtils.java [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+
+                // Get hold of the response entity
+                HttpEntity entity = response.getEntity();
+                // If the response does not enclose an entity, there is no need
+                // to worry about connection release
+
+                if (entity != null) {
+
+                    InputStream instream = entity.getContent();
+                    String result= convertStreamToString(instream);
```
Comment (thevery) [inline pohularPhotos/src/main/java/ru/skipor/Utils/HTTPUtils.java:69]: use `EnititUtils.toString()` instead

File: pohularPhotos/src/main/java/ru/skipor/Utils/InternalStorageUtils.java [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+public class InternalStorageUtils {
+    private final static String TAG = "InternalStorageUtils";
+
+    public static synchronized void saveBitmap(Context context, String fileName, Bitmap bitmapImage) {
+
+        FileOutputStream fileOutputStream;
+
+        deleteFileIfExists(context, fileName);
+        try {
+            fileOutputStream = context.openFileOutput(fileName, Context.MODE_PRIVATE);
```
Comment (thevery) [inline pohularPhotos/src/main/java/ru/skipor/Utils/InternalStorageUtils.java:28]: download-save-decode is much more efficient than download-decode-save (actually re-encode)

File: pohularPhotos/src/main/java/ru/skipor/popularPhotos/MainActivity.java [hunk 3]
```diff
@@ -0,0 +1,378 @@
...
+
+        Display display = getWindowManager().getDefaultDisplay();
+        Point size = new Point();
+        display.getSize(size);
+
+        int windowWidth = size.x;
+        minScreenDimension = Math.min(size.x, size.y);
+
+        maxSmallImageSize = Math.max(getLandscapeColumnWidth(Math.max(size.x, size.y)), getPortraitColumnWidth(Math.min(size.x, size.y)));
+        if (getResources().getConfiguration().orientation == Configuration.ORIENTATION_PORTRAIT) {
```
Comment (thevery) [inline pohularPhotos/src/main/java/ru/skipor/popularPhotos/MainActivity.java:71]: use `layout` (or `values`) and `layout-land` (`values-land`) resource folders instead

File: pohularPhotos/src/main/java/ru/skipor/popularPhotos/MainActivity.java [hunk 4]
```diff
@@ -0,0 +1,378 @@
...
+            super.onProgressUpdate(values);
+            adapter.add(values[0]);
+        }
+
+        @Override
+        protected void onCancelled(Void aVoid) {
+            Log.d(TAG, "onCancelled");
+            super.onCancelled(aVoid);
+            if(!HTTPUtils.checkConnection(MainActivity.this)) {
+                Toast.makeText(MainActivity.this,"No internet connection", Toast.LENGTH_SHORT).show();
```
Comment (thevery) [inline pohularPhotos/src/main/java/ru/skipor/popularPhotos/MainActivity.java:167]: strings.xml

File: pohularPhotos/src/main/java/ru/skipor/popularPhotos/PopularPhotoDataProvider.java [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+
+    public PhotoUrlsList getPopularPhotos() throws HttpException {
+
+        try {
+            PhotoUrlsList urlsList = new PhotoUrlsList();
+
+            final String content = HTTPUtils.getContent(QUERY_URL);
+            Log.d(TAG, content);
+            JSONObject response = new JSONObject(content);
+            JSONArray photosArray = response.getJSONArray("entries");
```
Comment (thevery) [inline pohularPhotos/src/main/java/ru/skipor/popularPhotos/PopularPhotoDataProvider.java:41]: `"entries"` should be const

**Comment by thevery** [comment]
wrong launcher icon size

**Comment by thevery** [comment]
20-8-2=10

File: pohularPhotos/src/main/java/ru/skipor/popularPhotos/PopularPhotoDataProvider.java (hunk 5 above)
Comment (skipor) [inline pohularPhotos/src/main/java/ru/skipor/popularPhotos/PopularPhotoDataProvider.java:41]: Too much const in class, and this line will not be changed in future, so in not const.

Comment (thevery) [inline pohularPhotos/src/main/java/ru/skipor/popularPhotos/PopularPhotoDataProvider.java:41]: `ORIGINAL_SIZE_TAG` is even never used, but you still created a const for it ;)

**Pull request closed without merging**